plugins/optimization-detective/detect.js

```diff
@@ -507,28 +540,70 @@ export default async function detect( {
 		);
 	} );
 
+	// Only proceed with submitting the URL Metric if viewport stayed the same size. Changing the viewport size (e.g. due
+	// to resizing a window or changing the orientation of a device) will result in unexpected metrics being collected.
+	if (
+		window.innerWidth !== urlMetric.viewport.width ||
+		window.innerHeight !== urlMetric.viewport.height
+	) {
+		if ( isDebug ) {
+			log(
+				'Aborting URL Metric collection due to viewport size change.'
+			);
+		}
+		return;
+	}
```

Comment:
Couldn't this be moved much further up in the logic, given we'll know whether the viewport has changed far earlier in this function? This could avoid doing quite a bit of "useless" processing as far as I can tell.

Reply:
Resolving in favor of https://github.com/WordPress/performance/pull/1712/files#r1866526591